OMCompiler/SimulationRuntime/fmi/export/buildproject/CMakeLists.txt.in

```diff
@@ -44,12 +44,20 @@ set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_SOURCE_DIR}/../binaries/${FMU_TARGET_PL
 set(CMAKE_INSTALL_LIBDIR ${CMAKE_INSTALL_PREFIX})
 set(CMAKE_INSTALL_BINDIR ${CMAKE_INSTALL_PREFIX})
 set(CMAKE_SHARED_LIBRARY_PREFIX "")
+
+# Set RPATH
+if(APPLE)
+  set(CMAKE_INSTALL_RPATH "@loader_path")
+else()
+  set(CMAKE_INSTALL_RPATH "$ORIGIN")
+endif()
+
 add_library(${FMU_NAME} SHARED
             ${FMU_RUNTIME_SOURCES}
             ${FMU_GENERATED_MODEL_SOURCES})
 
-
-target_link_libraries(${FMU_NAME} PRIVATE Threads::Threads)
+target_link_options(${FMU_NAME} PRIVATE -Wl,--no-undefined)
```

Comment:
This is added in CMake 3.13. Don't feel bad though, it is problematic anyway because the flag is not always the same between compilers (e.g., AppleClang will fail)

Reply:
I changed it to `LINKER:SHELL:--no-undefined"` which should work, as long as the linker knows `--no-undefined`.